src/ModelContextProtocol/ClientTokenProgress.cs

```diff
@@ -0,0 +1,28 @@
+using ModelContextProtocol.Client;
+using ModelContextProtocol.Protocol.Messages;
+
+namespace ModelContextProtocol;
+
+internal sealed class ClientTokenProgress(IMcpClient client, ProgressToken? progressToken)
+    : IProgress<ProgressNotificationValue>
+{
+    /// <inheritdoc />
+    public void Report(ProgressNotificationValue value)
+    {
+        if (progressToken is null) return;
```

Comment:
Assuming we do the singleton thing, then this line can be removed and progressToken can be changed to be non-nullable.